corehq/apps/locations/resources/v0_6.py

```diff
@@ -120,3 +122,15 @@ def _validate_new_parent(self, domain, location, parent):
         if location.location_type not in parent_allowed_types:
             raise BadRequest(_("Parent cannot have children of this location's type."))
         self._validate_unique_among_siblings(location, location.name, parent)
+
+    @atomic
+    def patch_list(self, request, **kwargs):
+        def create_or_update(bundle, **kwargs):
+            if 'location_id' in bundle.data and SQLLocation.objects.filter(
+                    location_id=bundle.data['location_id'], domain=kwargs['domain']).exists():
+                bundle = self.obj_update(bundle, **kwargs)
+            else:
+                bundle = self.obj_create(bundle, **kwargs)
```

Comment:
If the user passes in `location_id`, but it's not found (it's not real, it was deleted, or it belongs to another domain), this will call `obj_create`.  I think instead, unrecognized `location_id`s should trigger errors.